src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java

```diff
@@ -0,0 +1,394 @@
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.common.ConsoleNotifier;
+import com.github.tomakehurst.wiremock.junit.WireMockClassRule;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import org.apache.commons.codec.Charsets;
+import org.assertj.core.util.Strings;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.cloud.contract.wiremock.WireMockSpring;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.bulkscancore.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.bulkscancore.domain.ExceptionCaseData;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CcdCallbackControllerTest {
+
+    private static final String AUTHORIZATION_HEADER_KEY = "Authorization";
+
+    private static final String SERVICE_AUTHORIZATION_HEADER_KEY = "ServiceAuthorization";
+
+    private static final String USER_AUTH_TOKEN = "Bearer TEST_USER_AUTH_TOKEN";
+
+    private static final String SERVICE_AUTH_TOKEN = "TEST_SERVICE_AUTH";
+
+    private static final String USER_TOKEN_WITHOUT_CASE_ACCESS = "USER_TOKEN_WITHOUT_CASE_ACCESS";
+
+    private static final String USER_ID = "1234";
+
+    private static final String START_EVENT_URL =
+        "/caseworkers/1234/jurisdictions/SSCS/case-types/Benefit/event-triggers/appealCreated/token";
+
+    private static final String SUBMIT_EVENT_URL =
+        "/caseworkers/1234/jurisdictions/SSCS/case-types/Benefit/cases?ignore-warning=true";
+
+    private static final String USER_ID_HEADER = "user-id";
+
+    private static final String KEY = "key";
+
+    private static final String VALUE = "value";
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    private int randomServerPort;
+
+    @MockBean
+    private AuthTokenValidator authTokenValidator;
+
+    @ClassRule
+    public static WireMockClassRule ccdServer = new WireMockClassRule(WireMockSpring.options().port(4000)
+        .bindAddress("localhost").notifier(new ConsoleNotifier(true)));
```

Comment:
doesn't it bind to localhost by default?

Reply:
done